tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -426,56 +572,303 @@ def get_fd_counts():
-        time.sleep(lb_stress_period
-                   )  # Let the system receive traffic for a set time period
-        rand_consumer.shutdown()
-        rand_consumer.wait()
-        fast_producer.wait()
+            "SIMPLE_DRIVER",
+            (workload, OMBSampleConfigurations.UNIT_TEST_LATENCY_VALIDATOR),
+            node=bench_node,
+            worker_nodes=worker_nodes)
+        benchmark.start()
+        benchmark_time_min = benchmark.benchmark_time() + 5
+        benchmark.wait(timeout_sec=benchmark_time_min * 60)
+        benchmark.check_succeed()
+
+        self.free_preallocated_nodes()
+
+    @cluster(num_nodes=12, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_many_partitions_compacted(self):
+        self._test_many_partitions(compacted=True)
+
+    @cluster(num_nodes=12, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_many_partitions(self):
+        self._test_many_partitions(compacted=False)
+
+    @cluster(num_nodes=12, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_omb(self):
+        scale = ScaleParameters(self.redpanda, replication_factor=3)
+        self.redpanda.start(parallel=True)
+
+        # We have other OMB benchmark tests, but this one runs at the
+        # peak partition count.
+        self._run_omb(scale)
+
+    def _test_many_partitions(self, compacted):
+        """
+        Validate that redpanda works with partition counts close to its resource
+        limits.
+
+        This test should evolve over time as we improve efficiency and can reliably
+        run with higher partition counts.  It should roughly track the values we
+        use for topic_memory_per_partition and topic_fds_per_partition.
+
+        * Check topic can be created.
+        * Check leadership election succeeds for all partitions.
+        * Write in enough data such that an unlimited size fetch
+          would exhaust ram (check enforcement of kafka_max_bytes_per_fetch).
+        * Consume all the data from the topic
+
+        * Restart nodes several times (check that recovery works, and that the additional
+          log segments created by rolling segments on restart do not cause us
+          to exhaust resources.
+
+        * Run a general produce+consume workload to check that the system remains in
+          a functional state.
+
+        This test dynamically scales with the nodes provided.  It can run
+        on docker environments, but that is only for developers iterating
+        on the test itself: meaningful tests of scale must be done on
+        dedicated nodes.
+        """
+
+        # Scale tests are not run on debug builds
+        assert not self.debug_mode
+
+        replication_factor = 3
+
+        scale = ScaleParameters(self.redpanda, replication_factor)
+
+        # Run with one huge topic: this is the more stressful case for Redpanda, compared
+        # with multiple modestly-sized topics, so it's what we test to find the system's limits.
+        n_topics = 1
```

Comment:
Curious, do you know what makes this more stressful? The main thing I know of is the increased size of the metadata.

Reply:
Yes, that's it.

Although the comment is kind of wrong, what really makes the thing stressful is having clients hit all the partitions together, and with some of the traffic generators that only talk to one partition at a time, that means putting in one topic.